libass/ass_cache.c

```diff
@@ -17,6 +17,12 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
+// We have some asserts on highly-contended atomics in here,
+// so they're disabled if not requested
+#if !defined(DEBUG) && !defined(NDEBUG)
+#define NDEBUG 1
```

Comment:
This disables cache asserts in almost all builds, including our own CI builds. If this is kept, all or most of our CI builds should be changed to explicitly set `-DDEBUG`

Reply:
I think we should either do that, or explicitly put the asserts on newly-atomic values behind an `#ifdef DEBUG` or the like; the perf implications can be pretty major at times, and disabling them in almost all builds is deliberate.

Reply:
I think instead of locally redefining `NDEBUG`, it would be better to create a specialized time-critical assert function for use in performance-sensitive contexts. It's not exclusive to the cache system, potential tiled rendering can benefit from it too, for example.

Reply:
Went ahead and did this (piggybacking on `DEBUG_LEVEL`).